src/pd/retry.rs

```diff
@@ -75,15 +76,22 @@ impl<Cl> RetryClient<Cl> {
 
 macro_rules! retry_core {
     ($self: ident, $tag: literal, $call: expr) => {{
+        #[cfg(feature = "prometheus")]
         let stats = pd_stats($tag);
         let mut last_err = Ok(());
         for _ in 0..LEADER_CHANGE_RETRY {
             let res = $call;
 
+            #[cfg(feature = "prometheus")]
             match stats.done(res) {
                 Ok(r) => return Ok(r),
                 Err(e) => last_err = Err(e),
             }
+            #[cfg(not(feature = "prometheus"))]
```

Comment:
It looks a little too invasive to me.

How about like this:

Define another `DummyRequestStats` with no member and empty methods. Then change `pd_stats` and `tikv_stats` to return coresponding object accroding to the feature.

Reply:
Sounds good. I'll get to it this evening.